### PR TITLE
Callable Syntax

### PR DESCRIPTION
**Pull request by maxfischer2781**
This PR adds support for ``(A) -> R`` callable syntax. Changes include:

* [x] added ``Callable``, ``Parameter`` and ``Signature`` representations to IR,
* [x] extended grammar to read ``(A, ...) -> R`` style steno types matching Python 3.8,
* [x] extended grammar to read ``(...) -> R`` special case,
* [x] support all ``Callable`` cases with ``.backend.typing.normalize``,
* [x] extended grammar to read ``Callable`` types,
* [x] added unit tests for new features,

The grammar and IR represents the *entire* [Python 3.8 call grammar](https://docs.python.org/3/reference/compound_stmts.html#function-definitions). This far exceeds what ``typing.Callable`` can represent, and will require #29 (or similar) to emit again. Since there is no distinction between ``Callable`` and ``Protocol`` in stenotype syntax itself, implementing a single IR/grammar straight away seemed less hacky.

### CODE REVIEW

**Comment by maxfischer2781** [comment]
As it turns out, ``typing.Callable`` is *absolutely* not like ``typing.Generic`` *at all*. Representing the former with the latter is a hack one way or another.

I've added an explicit ``Callable`` IR type, which vastly simplifies handling (i.e. normalize and unparse). We should already cover all valid ``Callable`` forms as steno types:
```
$ venv/bin/stenotype '(*_) -> R'
typing.Callable[..., R]
$ venv/bin/stenotype '(A, B, C) -> R'
typing.Callable[[A, B, C], R]
$ venv/bin/stenotype '(A, B, /, C) -> R'
typing.Callable[[A, B, C], R]
```
Note that we cannot *parse* ``Callable[..., R]`` and ``Callable[[A, B, C], R]`` yet. However, this should be doable due to ``typing.Callable`` being *absolutely* not like ``typing.Generic`` *at all*.

File: tests/backend/test_typing.py [hunk 1]
```diff
@@ -77,6 +146,22 @@ def test_shorthands(typing_type, element):
...
+def test_callables(typing_type, element):
+    assert typing_type == unparse(normalize(element))
+
+
+@pytest.mark.parametrize("typing_type, element", callable_signatures)
+def test_callable_signatures(typing_type, element):
+    assert typing_type == unparse(normalize(element))
+
+
+@pytest.mark.parametrize("element", protocol_signatures)
```
Comment (a-recknagel) [inline tests/backend/test_typing.py:159]: Do you want to solve the Protocol-fallback within this PR, or push it like this?

File: tests/backend/test_grammar.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import pytest
 
 from stenotype.backend import elements as ste
-from stenotype.backend.grammar import parse
+from stenotype.backend.grammar import parse, SIGNATURE
```
Comment (a-recknagel) [inline tests/backend/test_grammar.py]:
```suggestion
from stenotype.backend.grammar import parse
```

**Review by a-recknagel — DISMISSED** [review]
lgtm

File: tests/backend/test_typing.py (hunk 1 above)
Comment (maxfischer2781) [reply tests/backend/test_typing.py:159]: I'd prefer to keep them separate. Protocols/Declarations may need some restructuring of normalize/unparse that aren't strongly tied to callables.

**Review by a-recknagel — APPROVED** [review]

**Pull request merged**